### PR TITLE
Geometric test whether ray intersects sphere

### PR DESCRIPTION
**Pull request by ArminBecher**
Added this feature since I needed it myself for another project where I use Pyrr. Thought it makes sense to add it here. 
Function return list of intersection points (possible 0 to 2) calculated via quadratic equation solver added to misc.

### CODE REVIEW

**Comment by adamlwgriffiths** [comment]
There's a bit of syntax touching in there. Is that a PEP-8 formatter?
Most of it is fine, but there are a couple that are off:
https://github.com/adamlwgriffiths/Pyrr/pull/83/commits/6c21afde2789e533bc12a8e3603391556ac97df3#diff-a1fa0401c2195fa7c8fd9dce2c6c4000R77
https://github.com/adamlwgriffiths/Pyrr/pull/83/commits/6c21afde2789e533bc12a8e3603391556ac97df3#diff-c2907d153021da1cec3549a759ba733fR7

Admittedly the code isn't PEP8, this lib originates from before I knew of it.
It would be better if we could do a PEP8 separately. Would you mind reducing this to just the code changes?

Apart from that, looks good. Thanks for including tests.
If you can undo the syntax we can do the merge.

**Comment by adamlwgriffiths** [comment]
Created #84 to cover syntax changes.

**Comment by adamlwgriffiths** [comment]
Sorry been busy, hopefully I'll get some time to look soon.
Appreciate it's not a big change.

File: pyrr/utils.py [hunk 1]
```diff
@@ -66,11 +67,41 @@ def wrapper(*args, **kwargs):
...
+        no solution for the quadratic equation exists.
+    """
+    delta = b * b - 4 * a * c
+    if delta > 0:
+        # Two solutions
+        # See https://www.scratchapixel.com/lessons/3d-basic-rendering/minimal-ray-tracer-rendering-simple-shapes/ray-sphere-intersection
+        # Why not use simple form:
+        # s1 = (-b + math.sqrt(delta)) / (2 * a)
+        # s2 = (-b - math.sqrt(delta)) / (2 * a)
+        q = -0.5 * (b + math.sqrt(delta)) if b > 0 else -0.5 * (b - math.sqrt(delta))
```
Comment (adamlwgriffiths) [inline pyrr/utils.py]:
```suggestion
        q = -0.5 * (b + np.sqrt(delta)) if b > 0 else -0.5 * (b - np.sqrt(delta))
```

File: pyrr/utils.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 """Provides common utility functions.
 """
 import inspect
+import math
```
Comment (adamlwgriffiths) [inline pyrr/utils.py]:
```suggestion
```

File: pyrr/utils.py (hunk 1 above)
Comment (adamlwgriffiths) [reply pyrr/utils.py]: We can use numpy's own sqrt which removes the extra math import

File: pyrr/geometric_tests.py [hunk 3]
```diff
@@ -375,6 +387,7 @@ def sphere_does_intersect_sphere(s1, s2):
         return False
     return True
 
+
```
Comment (adamlwgriffiths) [inline pyrr/geometric_tests.py]: There are still a lot of superfluous pep8 changes here.
Would be good if this was isolated to the change itself.
These can be done as part of #84

**Comment by ArminBecher** [comment]
Sorry for very(!) late reply to this small issue. Branch should be clean now.

**Comment by adamlwgriffiths** [comment]
Nice man, thanks for coming back and doing that =)
Those changes aren't pointless, the code needs a spring clean, but I just want to keep things atomic.

**Comment by adamlwgriffiths** [comment]
We should add you to the contributors too.

**Comment by ArminBecher** [comment]
Thanks man. Of course, would be nice. Just add me. 
I will hopefully find the time to contribute more.

**Pull request merged**